docs/source/concepts/architecture/custom-types/datatree-ditto.rst

```diff
@@ -0,0 +1,150 @@
+xarray datasets. This allows users to store data types such as NumPy arrays or lists in a hierarchical tree format while
+preserving enough metadata for reversible conversion.
+
+The name "Ditto" is inspired by the Pokémon character that mimics others — similarly, this class mimics ``DataTree``,
+with added functionality for type handling.
+
+Usage Example::
+
+    >>> import numpy as np
+    >>> from your_module import DataTreeDitto
+    >>> arr = np.array([[1, 2], [3, 4]])
+    >>> dt = DataTreeDitto(arr)
+    >>> dt.ds.data.values.tolist()
+    [[1, 2], [3, 4]]
+    >>> dt.get_original().tolist()
+    [1, 2, 3]
+
+DataTreeDitto internally relies on xarray's DataTrees and metadata used for round-trip conversion of non-xobjects is
+stored in dataset attributes prefixed with ``_ditto_``.
+
+Custom converters can be registered using ``.register_converter()`` for full extensibility and some basic converters
+(e.g., for numpy nd-arrays) are provided.
+
+Using DataTreeDitto
+===================
+
+Because DataTreeDittos are xarray DataTrees,
+you can use DataTreeDitto to do anything you can do with an :class:`xarray.DataTree`:
+
+.. code-block:: python
+
+    >>> dtd = DataTreeDitto()
+    >>> dt = xarray.DataTree()
+    >>> isinstance(dt, xarray.DataTree) # A DataTree is a DataTree
+    True
+    >>> isinstance(dt, DataTreeDitto) # A DataTree is NOT a DataTreeDitto
+    False
+    >>> isinstance(dtd, DataTreeDitto) # A DataTreeDitto is a DataTreeDitto
+    True
+    >>> isinstance(dtd, xarray.DataTree) # A DataTreeDitto IS a DataTree
+    True
+
+Automatic Conversion of Children
+--------------------------------
+
+.. code-block:: python
+
+   tree["child_1"] = np.array([10, 20])
+   tree["child_2"] = {"a": "b"}  # Will raise TypeError unless converter is registered
+
+   # Get child as DataTreeDitto instance:
+   child = tree["child_1"]
+   assert isinstance(child, DataTreeDitto)
+
+Hierarchical Tree Structure:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+   tree["sub"]["nested"] = np.ones((2,))
+   print(tree.to_datatree())  # Returns standard xarray.DataTree version for compatibility.
+
+Round-trip Recovery:
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: python
+
+   recovered = tree.get_original("sub/nested")
+   assert isinstance(recovered, np.ndarray)
+
+Advanced Usage: Custom Converters
+---------------------------------
+
+You can register custom converters for your own data types.
+
+Example: Handling Lists
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Suppose you want to allow lists to be stored in a tree node.
+
+Define conversion functions:
+
+.. code-block:: python
+
+   def list_to_dataset(lst, name="data", **kwargs):
+       return DataTreeDitto._numpy_to_dataset(np.array(lst), name=name)
+
+   def dataset_to_list(ds: xr.Dataset):
+       return ds[list(ds.data_vars)[0]].values.tolist()
+
+Register the converter:
+
+.. code-block:: python
+
+   DataTreeDitto.register_converter(list, list_to_dataset, dataset_to_list)
```

Comment:
Is the first argument here the type that is intended to be handled by the converter?

Reply:
Yep!